logbook-spring-boot-autoconfigure/src/main/java/org/zalando/logbook/autoconfigure/interceptors/LogbookClientHttpRequestInterceptor.java

```diff
@@ -0,0 +1,67 @@
+package org.zalando.logbook.autoconfigure.interceptors;
+
+import org.apache.http.ProtocolVersion;
+import org.apache.http.message.BasicHttpRequest;
+import org.apache.http.message.BasicHttpResponse;
+import org.apache.http.protocol.BasicHttpContext;
+import org.apache.http.protocol.HttpContext;
+import org.apiguardian.api.API;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.zalando.logbook.httpclient.LogbookHttpRequestInterceptor;
+import org.zalando.logbook.httpclient.LogbookHttpResponseInterceptor;
+
+import java.io.IOException;
+
+/**
+ * Uses {@link LogbookHttpResponseInterceptor} and {@link LogbookHttpResponseInterceptor} to log the start and
+ * completion of client http requests.
+ */
+@API(status = API.Status.EXPERIMENTAL)
+public class LogbookClientHttpRequestInterceptor implements ClientHttpRequestInterceptor {
```

Comment:
This implementation is now has a compile-time dependency to apache http-client, even if someone is using the RestTemplate with another HTTP client implementation.

A cleaner way would be to implement it completely separate and also to move it to its own module, e.g. `logbook-spring`.

Reply:
(For those users that are just using Spring, but not Spring Boot)